sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/EndToEndTests.java

```diff
@@ -695,7 +695,7 @@ public void decimalParse() {
         String query = "SELECT CAST('0.5' AS DECIMAL)";
         this.testConstantOutput(query, new DBSPZSetLiteral.Contents(
                 new DBSPTupleExpression(
-                        new DBSPDecimalLiteral(new DBSPTypeDecimal(CalciteObject.EMPTY, DBSPTypeDecimal.MAX_PRECISION, DBSPTypeDecimal.MAX_SCALE, false), new BigDecimal("0.5")))));
+                        new DBSPDecimalLiteral(new DBSPTypeDecimal(CalciteObject.EMPTY, DBSPTypeDecimal.MAX_PRECISION, 0, false), new BigDecimal("0.5")))));
```

Comment:
How can this literal have this type?
Something is wrong.
Perhaps we should add some checks in the DecimalLiteral constructor that the value fits in the type?
We can at least give a warning if the value does not fit.
Perhaps this works because the conversion is done at runtime?

Reply:
Casting to DECIMAL by default is `DECIMAL(MAX_PRECISION, 0)`. So, 0.5 with scale 0, will be 1. 
But I should change the expected output from `0. 5` to `1`.